l5kit/l5kit/evaluation/metrics.py

```diff
@@ -32,6 +27,28 @@ def neg_multi_log_likelihood(
     assert np.isfinite(confidences).all(), "invalid value found in confidences"
     assert np.isfinite(avails).all(), "invalid value found in avails"
 
+
+def neg_multi_log_likelihood(
+    gt: np.ndarray, pred: np.ndarray, confidences: np.ndarray, avails: np.ndarray
+) -> np.ndarray:
+    """
+    Compute a negative log-likelihood for the multi-modal scenario.
+    log-sum-exp trick is used here to avoid underflow and overflow, For more information about it see:
+    https://en.wikipedia.org/wiki/LogSumExp#log-sum-exp_trick_for_log-domain_calculations
+    https://timvieira.github.io/blog/post/2014/02/11/exp-normalize-trick/
+    https://leimao.github.io/blog/LogSumExp/
```

Comment:
Maybe we can add a bit more comments here? Ie, mention first the formula which is our loss, and then show the modified formula for stability?